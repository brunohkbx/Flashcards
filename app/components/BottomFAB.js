import React  from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import FAB from './FAB';
import { View } from 'react-native';

const Container = styled(View)`
  position: absolute;
  right: 16;
  bottom: 16;
`;

const BottomFAB = ({ handlePress }) => {
  return (
    <Container>
      <FAB handlePress={handlePress} />
    </Container>
  );
};

BottomFAB.propTypes = {
  handlePress: PropTypes.func.isRequired
};

export default BottomFAB;
