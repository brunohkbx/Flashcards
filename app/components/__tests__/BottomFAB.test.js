import React from 'react';
import { shallow } from 'enzyme';
import BottomFAB from '../BottomFAB';

describe('BottomFAB', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      handlePress: jest.fn()
    }, propOverrides)

    const wrapper = shallow(<BottomFAB {...defaultProps} />);

    return { wrapper };
  };

  it('renders properly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
