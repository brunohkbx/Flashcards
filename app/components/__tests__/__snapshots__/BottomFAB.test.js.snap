// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomFAB renders properly 1`] = `
<Styled(Component)>
  <FAB
    handlePress={[MockFunction]}
    primary={false}
  />
</Styled(Component)>
`;
